plugins/module_utils/vmware.py

```diff
@@ -1048,11 +1048,11 @@ def option_diff(options, current_options, truthy_strings_as_bool=True):
     for option_key, option_value in options.items():
         if truthy_strings_as_bool and is_boolean(option_value):
             option_value = VmomiSupport.vmodlTypes['bool'](is_truthy(option_value))
-        elif isinstance(option_value, int):
+        elif type(option_value) is int:
             option_value = VmomiSupport.vmodlTypes['int'](option_value)
-        elif isinstance(option_value, float):
+        elif type(option_value) is float:
```

Comment:
Would it be possible to write a test case for this change?

Reply:
Thanks for looking at this one.
I added the unit test for the option_diff method.